[Swift in WebKit] Continue work towards modularizing WebCore (part 1)

rr-codes opened the pull request:
#### dba962b2a56d30037022920c5467fe3e4e176512
<pre>
[Swift in WebKit] Continue work towards modularizing WebCore (part 1)
<a href="https://rdar.apple.com/165217840">rdar://165217840</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=302959">https://bugs.webkit.org/show_bug.cgi?id=302959</a>

Reviewed by Mike Wyrzykowski.

Fix some more module related issues before introducing the WebCore module.

* Source/WebCore/animation/AnimationTimeline.h:
* Source/WebCore/platform/animation/AcceleratedTimeline.h:
* Source/WebCore/platform/animation/ProgressResolutionData.h:
* Source/WebCore/Modules/fetch/FetchBodyConsumer.h:
* Source/WebCore/Modules/Model/InternalAPI/DDMaterialDescriptor.h:
* Source/WebCore/Modules/Model/InternalAPI/DDTextureDescriptor.h:
* Source/WebCore/Modules/Model/InternalAPI/DDUpdateTextureDescriptor.h:
* Source/WebCore/platform/TextRecognitionResult.h:
* Source/WebCore/style/values/backgrounds/StyleLineWidth.h:

- Fix quoted header imports

* Source/WebCore/platform/graphics/angle/ANGLEUtilities.h:
* Source/WebCore/platform/network/cf/ResourceRequest.h:
* Source/WebCore/platform/network/BlobDataFileReference.h:
* Source/WebCore/platform/network/FormData.h:
* Source/WebCore/platform/network/ResourceRequestBase.h:
* Source/WebCore/platform/network/ResourceResponseBase.h:

- Add missing includes

* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.mm:
(WebCore::SourceBufferParserAVFObjC::logChannel const):

- Fix `logChannel` definition to be properly defined in the implementation file like all other definitions of it.

* Source/WebCore/platform/mac/WebCoreObjCExtras.h:

- Allow this file to be included anywhere, just guard the contents

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

- Make some private headers project, and vice versa

* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::attachmentFileWrapperClassSingleton const):
(API::PageConfiguration::setAttachmentFileWrapperClass):

- Do not rely on a (transitive) dependency on `Decoder.h` just for `ClassStructPtr`,
especially since it&apos;s never used in APIPageConfiguration in a non-objc context

* Source/WebKit/UIProcess/WebURLSchemeTask.h:

- Forward declare `WebProcessProxy` instead of including it, and instead include only
the headers that are actually needed.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

- Let Xcode sort some stuff, and also add some missing headers to the xcode project.

Canonical link: <a href="https://commits.webkit.org/303439@main">https://commits.webkit.org/303439@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9602043caff91c3e7c860d78a9ec62d34113337f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132416 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4911 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43459 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139931 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84373 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134286 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4929 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4670 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101228 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68484 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/5f192705-abd7-4e3a-a6d5-6ba4f9e7578a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135362 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3462 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118608 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82020 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/cddad0f4-c30a-49d2-ba6e-f512a193bc1c) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3345 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/1210 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83158 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/112192 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36725 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142583 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4581 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37314 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109604 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4663 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3950 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109783 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27820 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3468 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114882 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57857 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4635 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33236 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4467 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68086 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4726 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4592 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -7338,6 +7324,8 @@
 		070756D814239B4B00414161 /* JSTextTrackCue.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = JSTextTrackCue.h; sourceTree = "<group>"; };
 		070756D914239B4C00414161 /* JSTextTrackCueList.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; path = JSTextTrackCueList.cpp; sourceTree = "<group>"; };
 		070756DA14239B4E00414161 /* JSTextTrackCueList.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = JSTextTrackCueList.h; sourceTree = "<group>"; };
+		0708C2292ECFAAC00023A523 /* WebCore.modulemap */ = {isa = PBXFileReference; lastKnownFileType = "sourcecode.module-map"; path = WebCore.modulemap; sourceTree = "<group>"; };
```
remove these

rr-codes commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
undo

rr-codes commented on Source/WebCore/animation/AnimationTimeline.h:
```diff
@@ -32,8 +32,8 @@
 #if ENABLE(THREADED_ANIMATIONS)
-#include "AcceleratedTimeline.h"
-#include "TimelineIdentifier.h"
+#include <WebCore/AcceleratedTimeline.h>
```
@graouts this will soon be enforced at compile time, but in the meantime if you could remember to try to use angle includes instead of quoted includes for any new imports you add, that will be really appreciated!! thanks!

graouts replied:
Duly noted, sorry I haven't caught up on that just yet and made additional work for you.

rr-codes commented on Source/WebCore/Modules/Model/InternalAPI/DDTextureDescriptor.h:
```diff
@@ -25,7 +25,7 @@
 #pragma once
 
-#include "DDImageAsset.h"
+#include <WebCore/DDImageAsset.h>
```
@mwyrzykowski this will soon be enforced at compile time, but in the meantime if you could remember to try to use angle includes instead of quoted includes for any new imports you add, that will be really appreciated!! thanks!

rr-codes committed:
[Swift in WebKit] Continue work towards modularizing WebCore (part 1)
rdar://165217840
https://bugs.webkit.org/show_bug.cgi?id=302959

Reviewed by Mike Wyrzykowski.

Fix some more module related issues before introducing the WebCore module.

* Source/WebCore/animation/AnimationTimeline.h:
* Source/WebCore/platform/animation/AcceleratedTimeline.h:
* Source/WebCore/platform/animation/ProgressResolutionData.h:
* Source/WebCore/Modules/fetch/FetchBodyConsumer.h:
* Source/WebCore/Modules/Model/InternalAPI/DDMaterialDescriptor.h:
* Source/WebCore/Modules/Model/InternalAPI/DDTextureDescriptor.h:
* Source/WebCore/Modules/Model/InternalAPI/DDUpdateTextureDescriptor.h:
* Source/WebCore/platform/TextRecognitionResult.h:
* Source/WebCore/style/values/backgrounds/StyleLineWidth.h:

- Fix quoted header imports

* Source/WebCore/platform/graphics/angle/ANGLEUtilities.h:
* Source/WebCore/platform/network/cf/ResourceRequest.h:
* Source/WebCore/platform/network/BlobDataFileReference.h:
* Source/WebCore/platform/network/FormData.h:
* Source/WebCore/platform/network/ResourceRequestBase.h:
* Source/WebCore/platform/network/ResourceResponseBase.h:

- Add missing includes

* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.mm:
(WebCore::SourceBufferParserAVFObjC::logChannel const):

- Fix `logChannel` definition to be properly defined in the implementation file like all other definitions of it.

* Source/WebCore/platform/mac/WebCoreObjCExtras.h:

- Allow this file to be included anywhere, just guard the contents

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

- Make some private headers project, and vice versa

* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::attachmentFileWrapperClassSingleton const):
(API::PageConfiguration::setAttachmentFileWrapperClass):

- Do not rely on a (transitive) dependency on `Decoder.h` just for `ClassStructPtr`,
especially since it's never used in APIPageConfiguration in a non-objc context

* Source/WebKit/UIProcess/WebURLSchemeTask.h:

- Forward declare `WebProcessProxy` instead of including it, and instead include only
the headers that are actually needed.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

- Let Xcode sort some stuff, and also add some missing headers to the xcode project.

Canonical link: https://commits.webkit.org/303439@main
